File: Library/PaintedSkin/src/main/java/org/alee/component/skin/page/IWindowProxy.java
package org.alee.component.skin.page;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import org.alee.component.skin.executor.SkinElement;
import org.alee.component.skin.service.ISwitchThemeSkinObserver;

/**********************************************************
 *
 * @author: MY.Liu
 * @date: 2021/2/14
 * @description: xxxx
 *
 *********************************************************/
public interface IWindowProxy extends ISwitchThemeSkinObserver {
    /**
     * 绑定 LayoutInflate
     *
     * @param context {@link Context}
     */
    void bindLayoutInflate(Context context);

    /**
     * 动态添加允许换肤的View
     *
     * @param view     可以换肤的View
     * @param elements 动态皮肤属性
     */
    void addEnabledThemeSkinView(@NonNull View view, @NonNull SkinElement... elements);

    /**
     * 恢复
     */
    void onVisible();

    /**
     * 暂停
     */
    void Invisible();


}

File: Library/PaintedSkin-Kotlin/src/main/java/org/alee/component/skin/core/window/IWindowProxy.kt
package org.alee.component.skin.core.window

import android.content.Context
import android.view.View
import androidx.annotation.WorkerThread
import org.alee.component.skin.model.SkinAttribute
import org.alee.component.skin.util.WeakReference
import org.jetbrains.annotations.TestOnly

/**
 * 摘要
 *
 * <p> 详细描述
 *
 * @author MingYu.Liu
 * created in 2023/9/12
 *
 */
internal sealed interface IWindowProxy {
    /**
     * 当窗口被创建时调用
     * @param context Context
     */
    fun onWindowCreated(context: Context)

    /**
     * 当窗口可见时调用
     */
    fun onWindowVisible()

    /**
     * 当窗口不可见时调用
     */
    fun onWindowInvisible()

    /**
     * 当窗口被销毁时调用
     */
    fun onWindowDestroyed()

    /**
     * 添加需要换肤的View
     * @param view View [View]
     * @param skinAttributes Array<out SkinAttribute> [SkinAttribute]
     */
    fun addSkinnableView(@WeakReference view: View, vararg skinAttributes: SkinAttribute)

    @TestOnly
    @WorkerThread
    fun dump(): String
}
